File: qt/gen_qsizepolicy.go
package qt

import (
	"unsafe"
)

type QSizePolicy__PolicyFlag int

const (
	QSizePolicy__GrowFlag   QSizePolicy__PolicyFlag = 1
	QSizePolicy__ExpandFlag QSizePolicy__PolicyFlag = 2
	QSizePolicy__ShrinkFlag QSizePolicy__PolicyFlag = 4
	QSizePolicy__IgnoreFlag QSizePolicy__PolicyFlag = 8
)

type QSizePolicy__Policy int

const (
	QSizePolicy__Fixed            QSizePolicy__Policy = 0
	QSizePolicy__Minimum          QSizePolicy__Policy = 1
	QSizePolicy__Maximum          QSizePolicy__Policy = 4
	QSizePolicy__Preferred        QSizePolicy__Policy = 5
	QSizePolicy__MinimumExpanding QSizePolicy__Policy = 3
	QSizePolicy__Expanding        QSizePolicy__Policy = 7
	QSizePolicy__Ignored          QSizePolicy__Policy = 13
)

type QSizePolicy__ControlType int

const (
	QSizePolicy__DefaultType QSizePolicy__ControlType = 1
	QSizePolicy__ButtonBox   QSizePolicy__ControlType = 2
	QSizePolicy__CheckBox    QSizePolicy__ControlType = 4
	QSizePolicy__ComboBox    QSizePolicy__ControlType = 8
	QSizePolicy__Frame       QSizePolicy__ControlType = 16
	QSizePolicy__GroupBox    QSizePolicy__ControlType = 32
	QSizePolicy__Label       QSizePolicy__ControlType = 64
	QSizePolicy__Line        QSizePolicy__ControlType = 128
	QSizePolicy__LineEdit    QSizePolicy__ControlType = 256
	QSizePolicy__PushButton  QSizePolicy__ControlType = 512
	QSizePolicy__RadioButton QSizePolicy__ControlType = 1024
	QSizePolicy__Slider      QSizePolicy__ControlType = 2048
	QSizePolicy__SpinBox     QSizePolicy__ControlType = 4096
	QSizePolicy__TabWidget   QSizePolicy__ControlType = 8192
	QSizePolicy__ToolButton  QSizePolicy__ControlType = 16384
)

type QSizePolicy struct {
	h          uintptr
	isSubclass bool
}

// NewQSizePolicy constructs a new QSizePolicy object.
func NewQSizePolicy() *QSizePolicy {

	ret := newQSizePolicy(QSizePolicy_new())
	ret.isSubclass = true
	return ret
}

// NewQSizePolicy2 constructs a new QSizePolicy object.
func NewQSizePolicy2(horizontal Policy, vertical Policy) *QSizePolicy {

	ret := newQSizePolicy(QSizePolicy_new2(horizontal, vertical))
	ret.isSubclass = true
	return ret
}

// NewQSizePolicy3 constructs a new QSizePolicy object.
func NewQSizePolicy3(param1 *QSizePolicy) *QSizePolicy {

	ret := newQSizePolicy(QSizePolicy_new3(param1.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQSizePolicy4 constructs a new QSizePolicy object.
func NewQSizePolicy4(horizontal Policy, vertical Policy, typeVal ControlType) *QSizePolicy {

	ret := newQSizePolicy(QSizePolicy_new4(horizontal, vertical, typeVal))
	ret.isSubclass = true
	return ret
}

func (this *QSizePolicy) HorizontalPolicy() Policy {
	xxxxxxxxx
}

func (this *QSizePolicy) VerticalPolicy() Policy {
	xxxxxxxxx
}

func (this *QSizePolicy) ControlType() ControlType {
	xxxxxxxxx
}

func (this *QSizePolicy) SetHorizontalPolicy(d Policy) {
	QSizePolicy_SetHorizontalPolicy(this.h, d)
}

func (this *QSizePolicy) SetVerticalPolicy(d Policy) {
	QSizePolicy_SetVerticalPolicy(this.h, d)
}

func (this *QSizePolicy) SetControlType(typeVal ControlType) {
	QSizePolicy_SetControlType(this.h, typeVal)
}

func (this *QSizePolicy) ExpandingDirections() Orientation {
	return (Orientation)(QSizePolicy_ExpandingDirections(this.h))
}

func (this *QSizePolicy) SetHeightForWidth(b bool) {
	QSizePolicy_SetHeightForWidth(this.h, (bool)(b))
}

func (this *QSizePolicy) HasHeightForWidth() bool {
	return (bool)(QSizePolicy_HasHeightForWidth(this.h))
}

func (this *QSizePolicy) SetWidthForHeight(b bool) {
	QSizePolicy_SetWidthForHeight(this.h, (bool)(b))
}

func (this *QSizePolicy) HasWidthForHeight() bool {
	return (bool)(QSizePolicy_HasWidthForHeight(this.h))
}

func (this *QSizePolicy) OperatorEqual(s *QSizePolicy) bool {
	return (bool)(QSizePolicy_OperatorEqual(this.h, s.cPointer()))
}

func (this *QSizePolicy) OperatorNotEqual(s *QSizePolicy) bool {
	return (bool)(QSizePolicy_OperatorNotEqual(this.h, s.cPointer()))
}

func (this *QSizePolicy) HorizontalStretch() int {
	return (int)(QSizePolicy_HorizontalStretch(this.h))
}

func (this *QSizePolicy) VerticalStretch() int {
	return (int)(QSizePolicy_VerticalStretch(this.h))
}

func (this *QSizePolicy) SetHorizontalStretch(stretchFactor int) {
	QSizePolicy_SetHorizontalStretch(this.h, (int)(stretchFactor))
}

func (this *QSizePolicy) SetVerticalStretch(stretchFactor int) {
	QSizePolicy_SetVerticalStretch(this.h, (int)(stretchFactor))
}

func (this *QSizePolicy) RetainSizeWhenHidden() bool {
	return (bool)(QSizePolicy_RetainSizeWhenHidden(this.h))
}

func (this *QSizePolicy) SetRetainSizeWhenHidden(retainSize bool) {
	QSizePolicy_SetRetainSizeWhenHidden(this.h, (bool)(retainSize))
}

func (this *QSizePolicy) Transpose() {
	QSizePolicy_Transpose(this.h)
}

func (this *QSizePolicy) Transposed() *QSizePolicy {
	_goptr := newQSizePolicy(QSizePolicy_Transposed(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

File: qt/gen_qsizepolicy.cpp
// +build ignore

#include <QSizePolicy>
#include <qsizepolicy.h>
#include "gen_qsizepolicy.h"

#ifndef _Bool
#define _Bool bool
#endif

void _GUID_Delete(_GUID* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<_GUID*>( self );
	} else {
		delete self;
	}
}

void type_info_Delete(type_info* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<type_info*>( self );
	} else {
		delete self;
	}
}

QSizePolicy* QSizePolicy_new() {
	return new QSizePolicy();
}

QSizePolicy* QSizePolicy_new2(Policy horizontal, Policy vertical) {
	return new QSizePolicy(horizontal, vertical);
}

QSizePolicy* QSizePolicy_new3(QSizePolicy* param1) {
	return new QSizePolicy(*param1);
}

QSizePolicy* QSizePolicy_new4(Policy horizontal, Policy vertical, ControlType typeVal) {
	return new QSizePolicy(horizontal, vertical, typeVal);
}

Policy QSizePolicy_HorizontalPolicy(const QSizePolicy* self) {
	return self->horizontalPolicy();
}

Policy QSizePolicy_VerticalPolicy(const QSizePolicy* self) {
	return self->verticalPolicy();
}

ControlType QSizePolicy_ControlType(const QSizePolicy* self) {
	return self->controlType();
}

void QSizePolicy_SetHorizontalPolicy(QSizePolicy* self, Policy d) {
	self->setHorizontalPolicy(d);
}

void QSizePolicy_SetVerticalPolicy(QSizePolicy* self, Policy d) {
	self->setVerticalPolicy(d);
}

void QSizePolicy_SetControlType(QSizePolicy* self, ControlType typeVal) {
	self->setControlType(typeVal);
}

int QSizePolicy_ExpandingDirections(const QSizePolicy* self) {
	Qt::Orientations _ret = self->expandingDirections();
	return static_cast<int>(_ret);
}

void QSizePolicy_SetHeightForWidth(QSizePolicy* self, bool b) {
	self->setHeightForWidth(b);
}

bool QSizePolicy_HasHeightForWidth(const QSizePolicy* self) {
	return self->hasHeightForWidth();
}

void QSizePolicy_SetWidthForHeight(QSizePolicy* self, bool b) {
	self->setWidthForHeight(b);
}

bool QSizePolicy_HasWidthForHeight(const QSizePolicy* self) {
	return self->hasWidthForHeight();
}

bool QSizePolicy_OperatorEqual(const QSizePolicy* self, QSizePolicy* s) {
	return (*self == *s);
}

bool QSizePolicy_OperatorNotEqual(const QSizePolicy* self, QSizePolicy* s) {
	return (*self != *s);
}

int QSizePolicy_HorizontalStretch(const QSizePolicy* self) {
	return self->horizontalStretch();
}

int QSizePolicy_VerticalStretch(const QSizePolicy* self) {
	return self->verticalStretch();
}

void QSizePolicy_SetHorizontalStretch(QSizePolicy* self, int stretchFactor) {
	self->setHorizontalStretch(static_cast<int>(stretchFactor));
}

void QSizePolicy_SetVerticalStretch(QSizePolicy* self, int stretchFactor) {
	self->setVerticalStretch(static_cast<int>(stretchFactor));
}

bool QSizePolicy_RetainSizeWhenHidden(const QSizePolicy* self) {
	return self->retainSizeWhenHidden();
}

void QSizePolicy_SetRetainSizeWhenHidden(QSizePolicy* self, bool retainSize) {
	self->setRetainSizeWhenHidden(retainSize);
}

void QSizePolicy_Transpose(QSizePolicy* self) {
	self->transpose();
}

QSizePolicy* QSizePolicy_Transposed(const QSizePolicy* self) {
	return new QSizePolicy(self->transposed());
}

void QSizePolicy_Delete(QSizePolicy* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QSizePolicy*>( self );
	} else {
		delete self;
	}
}

